### PR TITLE
PRO-19 3/UI theme plumbing

### PR DESCRIPTION
**Pull request by stuartromanek**
This PR goes beyond the [scope of the ticket](https://linear.app/apostrophecms/issue/PRO-19/trash-filter-missing-radio-button-for-the-active-state)

- tees up a theme object that might contain various UI preferences by cascading through user and global preferences
- pushes it to the browser's `apos` object
- top level apos vue apps apply it to there wrapper, replacing the hardcoded version
- ✅ context menus now apply the theme, which solves the original bug of missing UI from the ticket

requesting review from both @abea and @boutell for their perspectives

### CODE REVIEW

File: modules/@apostrophecms/login/ui/apos/components/TheAposLogin.vue [hunk 1]
```diff
@@ -1,59 +1,62 @@
...
-                {{ context.env }}
-              </label>
-              <label class="apos-login__project apos-login__project-name">
-                {{ context.name }}
-              </label>
-              <label class="apos-login--error">
-                {{ error }}
-              </label>
-            </div>
+  <div ref="login">
```
Comment (stuartromanek) [inline modules/@apostrophecms/login/ui/apos/components/TheAposLogin.vue:2]: needed to wrap this in something .. an element's `$ref` are not available while it's hidden (pre transition) so adding it inside the transition was not working (and lost a bit of time figuring it)

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue [hunk 2]
```diff
@@ -456,6 +454,7 @@ export default {
   },
   async mounted() {
     window.apos.adminBar.height = this.$refs.adminBar.offsetHeight;
+    window.apos.util.addClass(this.$refs.adminBarOuter, `apos-theme--primary-${window.apos.ui.theme.primary}`);
```
Comment (boutell) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue:457]: Can't this be done with :class and a computed value?

File: modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue [hunk 3]
```diff
@@ -1,5 +1,8 @@
 <template>
-  <div :data-apos-area="areaId" class="apos-area apos-theme--primary-purple">
+  <div
+    :data-apos-area="areaId" class="apos-area"
+    ref="areaEditor"
```
Comment (boutell) [inline modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue:4]: Same question.

File: modules/@apostrophecms/busy/ui/apos/components/TheAposBusy.vue [hunk 4]
```diff
@@ -14,6 +14,7 @@ export default {
     };
   },
   mounted() {
+    window.apos.util.addClass(this.$refs.busy, `apos-theme--primary-${window.apos.ui.theme.primary}`);
```
Comment (boutell) [inline modules/@apostrophecms/busy/ui/apos/components/TheAposBusy.vue:17]: Hoping all of these can be made more vuetiful.

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue [hunk 5]
```diff
@@ -1,7 +1,5 @@
 <template>
-  <!-- TODO: Replace this and other hard-coded `apos-theme--primary-purple`
```
Comment (abea) [inline modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue:2]: 🚮 🏆

File: modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue [hunk 6]
```diff
@@ -172,6 +175,7 @@ export default {
     }
   },
   mounted() {
+    window.apos.util.addClass(this.$refs.areaEditor, `apos-theme--primary-${window.apos.ui.theme.primary}`);
```
Comment (abea) [inline modules/@apostrophecms/area/ui/apos/components/AposAreaEditor.vue:178]: > Same question. - TB

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue (hunk 2 above)
Comment (abea) [reply modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue:457]: I'm thinking these should be in `created` instead. Get in there earlier. Assuming `$refs` is there then.

Comment (abea) [reply modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue:457]: Actually, if it's computed we no longer need the `ref`.

File: modules/@apostrophecms/ui/ui/apos/scss/global/_theme.scss [hunk 7]
```diff
@@ -104,23 +99,23 @@
...
 }
-.apos-theme--primary-light-blue {
+.apos-theme--primary-aqua {
   @include apos-primary-mixin(#1eacc7);
 }
 .apos-theme--primary-sun {
   @include apos-primary-mixin(#f7a704);
 }
-.apos-theme--primary-money {
+.apos-theme--primary-green {
```
Comment (abea) [inline modules/@apostrophecms/ui/ui/apos/scss/global/_theme.scss:118]: 💸

**Review by abea — CHANGES_REQUESTED** [review]

**Review by stuartromanek — COMMENTED** [review]
how do we like this approach

File: modules/@apostrophecms/login/ui/apos/components/TheAposLogin.vue (hunk 1 above)
Comment (stuartromanek) [reply modules/@apostrophecms/login/ui/apos/components/TheAposLogin.vue]: gone now

File: modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue (hunk 2 above)
Comment (stuartromanek) [reply modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue]: now a computed via a theme mixin

File: modules/@apostrophecms/busy/ui/apos/components/TheAposBusy.vue [hunk 8]
```diff
@@ -1,18 +1,32 @@
...
       busyCount: 0
     };
   },
+  computed: {
+    classes() {
+      const classes = [];
+      if (this.busy) {
+        classes.push('is-busy');
+      }
+      return [].concat(classes, this.themeClass);
```
Comment (abea) [inline modules/@apostrophecms/busy/ui/apos/components/TheAposBusy.vue:27]: Could just be `classes.concat(this.themeClasse)`, no?

**Review by abea — CHANGES_REQUESTED** [review]
Much better. One note on one where the concat is weird.

File: modules/@apostrophecms/busy/ui/apos/components/TheAposBusy.vue (hunk 8 above)
Comment (stuartromanek) [reply modules/@apostrophecms/busy/ui/apos/components/TheAposBusy.vue]: simplified

**Review by abea — APPROVED** [review]

**Review by boutell — APPROVED** [review]

**Pull request merged**